ini-file-parser/src/main/java/com/baeldung/inifileparser/CommonsParser.java

```diff
@@ -0,0 +1,50 @@
+package com.baeldung.inifileparser;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.configuration2.INIConfiguration;
+import org.apache.commons.configuration2.SubnodeConfiguration;
+
+public class CommonsParser {
+
+    private File fileToParse;
+
+    public CommonsParser(File iniFile) {
+        this.fileToParse = iniFile;
```

Comment:
This is a bad practice. Have static methods, rather than a request-scope private field. Every request would need to create a new object... an object will just one function. It's not an object. It's just a function.